engine/dlib/src/dmsdk/dlib/log.h

```diff
@@ -224,49 +226,92 @@ void LogInternal(Severity severity, const char* domain, const char* format, ...)
 #define dmLogOnceFatal(format, args... ) dmLogOnceInternal(dmLogFatal, format, ## args )
 #endif
 
-#endif // NDEBUG
-
 /*# dmLog:LogListener callback typedef
  *
  * dmLog listener function type. Provides all logs from dmLog* functions and print/pprint Lua functions.
  * Used with dmLogRegisterListener() and dmLogUnregisterListener()
  *
  * @typedef
  * @name dmLog:LogListener
- * @param severity [type:dmLog::Severity]
+ * @param severity [type:LogSeverity]
  * @param domain [type:const char*]
  * @param formatted_string [type:const char*] null terminated string
  */
-typedef void (*LogListener)(Severity severity, const char* domain, const char* formatted_string);
+typedef void (*FLogListener)(LogSeverity severity, const char* domain, const char* formatted_string);
+
+/*# register a log listener.
+ *
+ * Registers a log listener.
+ * This listener receive logs even in release bundle.
+ *
+ * @name dmLogRegisterListener
+ * @param listener [type:FLogListener]
+ */
+void dmLogRegisterListener(FLogListener listener);
+
+/*# unregister a log listener.
+ *
+ * Unregisters a log listener.
+ *
+ * @name dmLogUnregisterListener
+ * @param [type:FLogListener] listener
+ */
+void dmLogUnregisterListener(FLogListener listener);
+
+/*# set log system severity level.
+ *
+ * set log system severity level.
+ *
+ * @name dmLogSetLevel
+ * @param [type:dmLog::Severity] severity
+ */
+void dmLogSetLevel(LogSeverity severity);
+
+
+#endif // NDEBUG
+
+#ifdef __cplusplus
+} // extern "C"
+#endif
+
+#ifdef __cplusplus
+
+namespace dmLog
+{
 
 /*# register dmLog listener.
  *
  * Registers a dmLog listener.
- * This listener recieve logs even in release bundle.
+ * This listener receive logs even in release bundle.
  *
+ * @note deprecated. Use dmLogRegisterListener instead.
  * @name dmLog::RegisterLogListener
- * @param listener [type:dmLog::LogListener]
+ * @param listener [type:FLogListener]
  */
-void RegisterLogListener(LogListener listener);
+void RegisterLogListener(FLogListener listener);
```

Comment:
Hopefully, there should be no issue with the typedef changing name, and bneing moved outside of the scope.
It should still be the same function signature.